### PR TITLE
POC for a mobile nutrition game

### PR DESCRIPTION
**Pull request by alexfauquette**
The nutrition game adapted to mobile.

# Objective

Make it more friendly to fill nutritional information

# How

- Allow in few click to select the appropriate image
- Allow to zoom/move in the image
- Ask one line at the time

# TODO

- [x] Use new AI to extract information
- [x] All the components are in the window (no need to scroll)
- [x] Smooth way to zoom and move in the picture

# Warning

This is a Proof of Concept, no API call is made (except to fetch data)

[The netlify link](https://deploy-preview-196--gifted-lalande-686eef.netlify.app/nutritionsWIP)

image

### CODE REVIEW

**Comment by teolemon** [comment]
- [x] The "Validate crop" button should be sticky at the bottom of the screen, with a big z-index so that it's always visible
- [x] As you already mentionned, the zoom behaviour is very distressing on desktop (I'm using a trackpad, and it spins out of control quickly)
- [ ] "Voir" and "Modifier" could be changed to "See product" and "Edit product" for added clarity it won't have impact on theannotation
- [x] "glucides" translation is incorrect, missing s
- [x] Does the modal that appears after clicking supprimer appear in some other cases. 
- [x] Be clearer about Supprimer: Unselect nutrition image or Change nutrition image

**Comment by teolemon** [comment]
Screenshots: https://photos.app.goo.gl/mCYr5kxZhkajUKfG8

**Comment by teolemon** [comment]
- [ ] Reword "Utilise Crop"
- [ ] I got an example without salt, fiber, but with Vitamin D (out of scope for now)

It works well on mobile 👍

File: src/components/Helper.vue [hunk 1]
```diff
@@ -78,7 +78,7 @@ Object.keys(explanations).forEach((urlKey) => {
 
 const getHelpInformations = (currentPath, translatedExplanations) => {
   let currentPathKey = currentPath.slice(1);
-  if (!(currentPathKey in translatedExplanations)) {
+  if (Object.keys(translatedExplanations).findIndex(currentPathKey) === -1) {
```
Comment (VaiTon) [inline src/components/Helper.vue]:
```suggestion
  if (!Object.keys(translatedExplanations).includes(currentPathKey)) {
```
Why don't we use includes?

Comment (alexfauquette) [reply src/components/Helper.vue]: Because I forget the existence of this method ;)

**Comment by alexfauquette** [comment]
J'ai complètement modifié la logique, car je trouvais le chemin utilisateur trop long. Maintenant, il y a 2 parties distinctes :
 - nutritionsWIP1 qui permet uniquement de cropper une image. Il faut ensuite indiquer si il s'agit d'un texte ou d'un tableau
 - nutritionsWIP2 permet de remplir les tableaux nutritionnel. Apparaissent (quand il y en a) les 3 valeurs les plus crédibles d’après l'IA
 
 
image

image

A discuter : 
- comment gérer les nutriments non classiques (vitamines, ...)
- comment gérer les cas problématiques (pas d'image de nutrition dispo, image qui correspond à une creme pour le visage, ...)?
- pour l'instant le lien entre les deux pages se fait par une petite api sur un serveur perso

**Comment by teolemon** [comment]
Je verrais bien un bouton pour afficher les Puces pour les autres valeurs. Je suis tombé sur un truc où tout était décalé.

**Comment by alexfauquette** [comment]
### Warning
Maintenant l'app est branché à l'API (nutritionWIP2 update les valeurs nutritionelles)

Par contre, les images ne sont pas déselectionné ni croppé dans la base OFF. Mais je garde toutes les données donc on pour le faire plus tard

**Comment by alexfauquette** [comment]
Peek 06-06-2021 11-33

**Comment by alexfauquette** [comment]
### **Warning** 
The app is now connected to the OFF API so you can edit nutri table. And for now, there is no easy way to know which product you edited (the best I know is to look at https://world-fr.openfoodfacts.org/editeur/openfoodfacts-contributors)

### URL
https://deploy-preview-196--gifted-lalande-686eef.netlify.app/nutritionComputer

### new features :
 - lines of the table were add in the order user clicks them. Now, they are ordered from top to bottom (y coordinate of the associated OCR box)
 - you can move come back to a previous step
 - when filling values of the table, you can click on a line toselect the nutriment to fill (useful if there is some nutrients for wich value is ne readable)
 
 ### missing features
  - being able to add nutiment or value event if there is not associated OCR. For example, "Fat" is not detected by the OCR, I can click on the image and say: _here it's written "fat" dummy OCR algo_
  - energy is a problem, because it has multiple units, so it does not feat well in current workflow. The hack is to click two times on "energy" and first select energy_kj and second energy_kcal
  - for nutriment selection, the UI should not propose a nutriment that is already in the table

**Comment by alexfauquette** [comment]
killer feature: You can select the country or your username to filter images

image

**Pull request closed without merging**